stairway/src/main/java/bio/terra/stairway/FlightMap.java

```diff
@@ -22,74 +19,41 @@
 public class FlightMap {
   private Map<String, String> map;
 
+  /** Construct an empty flight map */
   public FlightMap() {
     map = new HashMap<>();
   }
 
   /**
-   * Alternate constructor, used by the DAO to re-create FlightMap from its serialized form.
+   * This constructor is used when Stairway re-creates the flight map from the database. It is not
+   * intended for client use.
    *
-   * @param inputList input list form of the input parameters
+   * @param map deserialized input map from database
    */
-  FlightMap(List<FlightInput> inputList) {
-    map = new HashMap<>();
-    for (FlightInput input : inputList) {
-      map.put(input.getKey(), input.getValue());
-    }
+  public FlightMap(Map<String, String> map) {
+    this.map = map;
   }
 
   /**
-   * Convert a flight map into the input list form. Used by the DAO to serialize the input
-   * parameters.
+   * Accessor for the contained map. Used for storing the map into the database. It is not intended
+   * for client use.
    *
-   * @return list of FlightInput
+   * @return contained map
    */
-  List<FlightInput> makeFlightInputList() {
-    ArrayList<FlightInput> inputList = new ArrayList<>();
-    for (Map.Entry<String, String> entry : map.entrySet()) {
-      inputList.add(new FlightInput(entry.getKey(), entry.getValue()));
-    }
-    return inputList;
-  }
-
-  /**
-   * Depending on what version of code a Flight log was written with, we may have a JSON Map, a
-   * {@code List<FlightInput>}, both, or neither at deserialization time. This method is used to
-   * generate a FlightMap based on what was contained in the database.
-   *
-   * <p>Flights written with:
-   *
-   * <ul>
-   *   <li>Version < 0.0.50: Only json is valid, inputList is always empty.
-   *   <li>Version 0.0.50 - 0.0.60: json and inputList both valid and must be consistent with one
-   *       another.
-   *   <li>Version >= 0.0.61: json always null, inputList is always valid (possibly empty).
-   * </ul>
-   *
-   * @param inputList Entries in flightworking for a given Flight log. May be empty if Flight
-   *     pre-existed flightworking table, or there are no working parameters.
-   * @param json Map of working entries for a given Flight log in JSON. May be NULL.
-   */
-  static FlightMap create(List<FlightInput> inputList, @Nullable String json) {
-
-    // If we have an empty list AND there is valid JSON, this predates the flightworking table...
-    // use the json in this case.
-    if (inputList.isEmpty() && json != null) {
-      FlightMap map = new FlightMap();
-      map.fromJson(json);
-      return map;
-    }
-
-    // Otherwise just use the input list (even if it's empty).
-    return new FlightMap(inputList);
```

Comment:
Removed and removed getMap.

Reply:
I still need `getMap` that exposes the map inside of the FlightMap. I don't see a good way around that.
I tried making a subclass that could then expose the map, but I can't cast to it from the FlightMap.
Any thoughts on how to get rid of it?

Reply:
I was just thinking create an empty `Map<String, String>`, copy `map` contents into it, and return it, to avoid letting a caller accidentally mutate the `FlightMap` internal state (unless that is actually the intent?).

Reply:
Even easier: I return `Collections.unmodifiableMap(map)` and don't have to copy the data.

Reply:
Thanks for getting me on the right track!